google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/GoogleCloudStorageModel.scala

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.storage
+
+import akka.annotation.InternalApi
+import akka.http.scaladsl.model.ContentType
+
+import scala.collection.immutable.Seq
+
+object GoogleCloudStorageModel {
+
+  /**
+   * Represents an object within Google Cloud Storage.
+   *
+   * @param kind The kind of item this is, for objects, this is always storage#object
+   * @param id The ID of the object, including the bucket name, object name, and generation number
+   * @param name The name of the object
+   * @param bucket The name of the bucket containing this object
+   * @param generation The content generation of this object, used for object versioning
+   * @param contentType Content-Type of the object data, if an object is stored without a Content-Type, it is served as application/octet-stream
+   * @param size Content-Length of the data in bytes
+   * @param etag HTTP 1.1 Entity tag for the object.
+   * @param md5Hash MD5 hash of the data; encoded using base64
+   * @param crc32c CRC32c checksum, encoded using base64 in big-endian byte order
+   * @param mediaLink Media download link
+   */
+  final class StorageObject private (
+      val kind: String,
+      val id: String,
+      val name: String,
+      val bucket: String,
+      val generation: Long,
+      val contentType: ContentType,
+      val size: Long,
+      val etag: String,
+      val md5Hash: String,
+      val crc32c: String,
+      val mediaLink: String
+  ) {
+
+    /** Java API */
+    def getKind: String = kind
+
+    /** Java API */
+    def getId: String = id
+
+    /** Java API */
+    def getName: String = name
+
+    /** Java API */
+    def getBucket: String = bucket
+
+    /** Java API */
+    def getGeneration: Long = generation
+
+    /** Java API */
+    def getContentType: akka.http.javadsl.model.ContentType = contentType.asInstanceOf[ContentType]
+
+    /** Java API */
+    def getSize: Long = size
+
+    /** Java API */
+    def getEtag: String = etag
+
+    /** Java API */
+    def getMd5Hash: String = md5Hash
+
+    /** Java API */
+    def getCrc32c: String = crc32c
+
+    /** Java API */
+    def getMediaLink: String = mediaLink
```

Comment:
Looks like you made good use of [KazeClass](https://github.com/ktoso/kaze-class).
I don't think the Java access methods are needed here.